pkg/neotest/basic.go

```diff
@@ -0,0 +1,268 @@
+	"github.com/nspcc-dev/neo-go/pkg/vm/emit"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+// Executor is a wrapper over chain state.
+type Executor struct {
+	Chain         blockchainer.Blockchainer
+	Committee     *wallet.Account
+	CommitteeHash util.Uint160
+	Contracts     map[string]*Contract
+}
+
+// NewExecutor creates new executor instance from provided blockchain and committee.
+func NewExecutor(t *testing.T, bc blockchainer.Blockchainer, committee *wallet.Account) *Executor {
+	require.Equal(t, 1, len(bc.GetConfig().StandbyCommittee))
+
+	return &Executor{
+		Chain:         bc,
+		Committee:     committee,
+		CommitteeHash: committee.Contract.ScriptHash(),
+		Contracts:     make(map[string]*Contract),
+	}
+}
+
+// TopBlock returns block with the highest index.
+func (e *Executor) TopBlock(t *testing.T) *block.Block {
+	b, err := e.Chain.GetBlock(e.Chain.GetHeaderHash(int(e.Chain.BlockHeight())))
+	require.NoError(t, err)
+	return b
+}
+
+// NativeHash returns native contract hash by name.
+func (e *Executor) NativeHash(t *testing.T, name string) util.Uint160 {
+	h, err := e.Chain.GetNativeContractScriptHash(name)
+	require.NoError(t, err)
+	return h
+}
+
+// NewUnsignedTx creates new unsigned transaction which invokes method of contract with hash.
+func (e *Executor) NewUnsignedTx(t *testing.T, hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	w := io.NewBufBinWriter()
+	emit.AppCall(w.BinWriter, hash, method, callflag.All, args...)
+	require.NoError(t, w.Err)
+
+	script := w.Bytes()
+	tx := transaction.New(script, 0)
+	tx.Nonce = nonce()
+	tx.ValidUntilBlock = e.Chain.BlockHeight() + 1
+	return tx
+}
+
+// NewTx creates new transaction which invokes contract method.
+// Transaction is signed with signer.
+func (e *Executor) NewTx(t *testing.T, signer interface{},
+	hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	tx := e.NewUnsignedTx(t, hash, method, args...)
+	return e.SignTx(t, tx, -1, signer)
+}
+
+// SignTx signs a transaction using provided signers.
+// signers can be either *wallet.Account or []*wallet.Account.
+func (e *Executor) SignTx(t *testing.T, tx *transaction.Transaction, sysFee int64, signers interface{}) *transaction.Transaction {
+	switch s := signers.(type) {
+	case *wallet.Account:
+		tx.Signers = append(tx.Signers, transaction.Signer{
+			Account: s.Contract.ScriptHash(),
+			Scopes:  transaction.Global,
+		})
+		addNetworkFee(e.Chain, tx, s)
+		addSystemFee(e.Chain, tx, sysFee)
+		require.NoError(t, s.SignTx(e.Chain.GetConfig().Magic, tx))
+	case []*wallet.Account:
+		for _, acc := range s {
+			tx.Signers = append(tx.Signers, transaction.Signer{
+				Account: acc.Contract.ScriptHash(),
+				Scopes:  transaction.Global,
+			})
+		}
+		for _, acc := range s {
+			addNetworkFee(e.Chain, tx, acc)
```

Comment:
But why? Could we do just
```
for _, acc := range s {
	tx.Signers = append(tx.Signers, transaction.Signer{
		Account: acc.Contract.ScriptHash(),
		Scopes:  transaction.Global,
	})
}
addNetworkFee(e.Chain, tx, s)
```
where `addNetworkFee` is the following:
```
func addNetworkFee(bc blockchainer.Blockchainer, tx *transaction.Transaction, s []*wallet.Account) {
	baseFee := bc.GetPolicer().GetBaseExecFee()
	size := io.GetVarSize(tx)
        for _, acc := range accs {
                netFee, sizeDelta := fee.Calculate(baseFee, acc.Contract.Script)
	        tx.NetworkFee += netFee
 	        size += sizeDelta
        }
	tx.NetworkFee += int64(size) * bc.FeePerByte()
}
```

Because otherwise, you add tx-size-fee for each signer instead of adding it once only, It may be nitpicking, but it's not correct.

Reply:
Fixed.